src/either.js

```diff
@@ -0,0 +1,124 @@
+'lang sweet.js';
+// @flow
+
+export class Either<L, R> {
+  static of(v: R): Either<L, R> {
+    return new Right(v);
+  }
+
+  static isLeft(e: Either<*, *>) {
+    return e instanceof Left;
+  }
+
+  static isRight(e: Either<*, *>) {
+    return e instanceof Right;
+  }
+  
+  map<C>(f: R => C): Either<L, C> {
+    throw new Error();
+  }
+
+  ap(either: Either<*, *>): Either<*, *> {
+    throw new Error();
+  }
+  
+  chain<C>(f: R => Either<L, C>): Either<L, C> {
+    throw new Error();
+  }
+
+  extend<C>(e: Either<L, R> => C): Either<L, C> {
+    throw new Error();
+  }
+  
+  bimap<C, D>(l: L => C, r: R => D): Either<C, D> {
+    throw new Error();
+  }
+  
+  either<C>(l: L => C, r: R => C): C {
+    throw new Error();
+  }
+
+  foreach(l: L => void, r: R => void): Either<L, R> {
+    throw new Error();
+  }
+}
+export class Left<L, R> extends Either<L, R> {
+  left: L;
+
+  constructor(left: L) {
+    super();
+    this.left = left;
+  }
+
+  map<C>(f: R => C): Either<L, C> {
+    return new Left(this.left);
+  }
+
+  ap(either: Either<*, *>): Either<*, *> {
+    return new Left(this.left);
+  }
+
+  chain<C>(f: R => Either<L, C>): Either<L, C> {
+    return new Left(this.left);
+  }
+
+  extend<C>(e: Either<L, R> => C): Either<L, C> {
+    return new Left(this.left);
+  }
+
+  bimap<C, D>(l: L => C, r: R => D): Either<C, D> {
+    return new Left(l(this.left));
+  }
+
+  either<C>(l: L => C, r: R => C): C {
+    return l(this.left);
+  }
+
+  foreach(l: L => void, r: R => void): Either<L, R> {
+    l(this.left);
+    return new Left(this.left);
+  }
+}
+
+export class Right<L, R> extends Either<L, R> {
+  right: R;
+
+  constructor(right: R) {
+    super();
+    this.right = right;
+  }
+
+  map<C>(f: R => C): Either<L, C> {
+    return new Right(f(this.right));
+  }
+
+  ap(either: any): Either<*, *> {
+    if (either instanceof Left) {
+      return either;
+    } else if (either instanceof Right && typeof this.right === 'function') {
+      return new Right(this.right(either.right));
```

Comment:
Not strictly Fantasy Land compatible. The argument order got flipped for the 1.0 release. Background https://github.com/fantasyland/fantasy-land/pull/145.

To make this API usable w/ Ramda/Sanctuary it would be a good idea to add `'fantasy-land/methodName'` prefixed methods as well.